### PR TITLE
Centre search input

### PR DESCRIPTION
**Pull request by mstriemer**
r? @ngokevin

### CODE REVIEW

File: src/media/css/elements--header.styl [hunk 1]
```diff
@@ -130,14 +130,25 @@ mkt-header-child {
     border: 0;
     border-radius: 5px;
     height: 70px;
-    padding: 15px 0 15px 20px;
-    position: relative;
     outline: none;
+    padding: 15px 40px;
```
Comment (ngokevin) [inline src/media/css/elements--header.styl]: Is the 40px for the icon? That might go into header--search since it's not part of the core element.

**Comment by ngokevin** [comment]
A bit of unittest failures

**Comment by ngokevin** [comment]
Placeholder can be centered more on mobile screen shot 2015-04-27 at 2 39 33 pm

File: src/media/css/elements--header.styl [hunk 2]
```diff
@@ -145,10 +156,13 @@ mkt-header-child {
...
     line-height: 40px;
     margin: 0;
+    max-width: 100%;
     opacity: .75;
-    padding: 15px 0;
+    overflow: hidden;
+    padding: 15px 40px;
     position: absolute;
     top: 50%;
+    transition: 250ms;
```
Comment (ngokevin) [inline src/media/css/elements--header.styl]: i'd make it a bit faster maybe

File: src/media/js/elements/header.js [hunk 3]
```diff
@@ -97,6 +97,24 @@ define('elements/header',
 
     el.MktHeaderChildElement = document.registerElement('mkt-header-child', {
         prototype: Object.create(HTMLElement.prototype, {
+            createdCallback: {
+                value: function() {
+                    if (this.isInput) {
+                        var input = this.input;
+                        var label = this.querySelector('label');
+                        var setInputWidth = function() {
+                            if (input.value) {
```
Comment (ngokevin) [inline src/media/js/elements/header.js]: maybe a ternary

**Comment by ngokevin** [comment]
looks mostly good

**Comment by mstriemer** [comment]
It's a bit off on mobile because it actually ends up being wider than the screen. It's 330px wide so at 320px it will be 5px off centre.

**Comment by mstriemer** [comment]
I'll remove the search icon and take some more screenshots, that should help.

File: src/media/css/elements--header.styl (hunk 2 above)
Comment (mstriemer) [inline src/media/css/elements--header.styl]: I made this transition when typing starts at `100ms`. Left the fade out at `250ms`. Should they both be `100ms`?

File: src/media/js/elements/header.js (hunk 3 above)
Comment (mstriemer) [inline src/media/js/elements/header.js]: Made it a ternery, I don't think it formats really well but it's the best I could come up with.

**Comment by mstriemer** [comment]
no-icon-search-mobile mov screen shot 2015-04-28 at 09 58 39
no-icon-search-desktop mov screen shot 2015-04-28 at 09 58 49

File: src/media/css/elements--header.styl [hunk 4]
```diff
@@ -9,6 +9,7 @@ $color--header--root-link = $greyscale-black;
 $header-child--input-width = $mobile-content-width + 100;
 $header--height = 60px;
 $header--transition = 'transform cubic-bezier(0.455, 0.03, 0.515, 0.955) 250ms';
+$search-input-padding = 15px 5px;
```
Comment (ngokevin) [inline src/media/css/elements--header.styl]: is this specific to search?

Comment (mstriemer) [inline src/media/css/elements--header.styl]: I don't think so. I'm assuming we want all inputs to behave this way.

**Comment by mstriemer** [comment]
Additional comments welcome but I've addressed everything here.

**Pull request merged**